sootup.java.core/src/main/java/sootup/java/core/JavaSootClass.java

```diff
@@ -53,50 +50,17 @@ public JavaClassType getType() {
   }
 
   /**
-   * Get all annotations on this class. If provided with a View, will also resolve all inherited
-   * annotations from super classes.
+   * Get all annotations that are directly attached to this class.
+   *
+   * <p>This includes "visible" and "invisible" annotations. Note that inherited annotations are not
+   * part of this iterable.
    *
-   * @param view
    * @return
    */
   @Nonnull
-  public Iterable<AnnotationUsage> getAnnotations(@Nonnull Optional<JavaView> view) {
+  public Iterable<AnnotationUsage> getAnnotations() {
     List<AnnotationUsage> annotationUsages = new ArrayList<>();
-
-    if (view.isPresent()) {
-      JavaView javaView = view.get();
-      if (this.getSuperclass().isPresent()) {
-
-        ClassType superClass = this.getSuperclass().get();
-
-        if (javaView.getClass(superClass).isPresent()) {
-          JavaSootClass superJavaSootClass = javaView.getClass(superClass).get();
-
-          Collection<AnnotationUsage> annos =
-              StreamSupport.stream(superJavaSootClass.getAnnotations(view).spliterator(), false)
-                  .filter(annotationUsage -> annotationUsage.getAnnotation().isInherited(view))
-                  .collect(Collectors.toList());
-
-          annotationUsages.addAll(annos);
-        }
-      }
-    }
-
     ((JavaSootClassSource) classSource).resolveAnnotations().forEach(annotationUsages::add);
```

Comment:
simplify. use collect(asList)

Reply:
lets cache it & return as Collections.unmodifiableList()?